moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/parallel_gripper_command_controller_handle.hpp

```diff
@@ -0,0 +1,214 @@
+      {
+        RCLCPP_ERROR(logger_, "ParallelGripperCommand expects a joint trajectory with one \
+                               point that specifies at least the position of joint \
+                               '%s', but insufficient positions provided.",
+                     trajectory.joint_trajectory.joint_names[idx].c_str());
+        return false;
+      }
+      cmd_state.position.push_back(trajectory.joint_trajectory.points[tpoint].positions[idx]);
+      // only set the velocity or effort if the user has specified a positive non-zero max value
+      if (max_velocity_ > 0.0)
+      {
+        cmd_state.velocity.push_back(max_velocity_);
+      }
+      if (max_effort_ > 0.0)
+      {
+        cmd_state.effort.push_back(max_effort_);
+      }
+    }
+    rclcpp_action::Client<control_msgs::action::ParallelGripperCommand>::SendGoalOptions send_goal_options;
+    // Active callback
+    send_goal_options.goal_response_callback =
+        [this](const rclcpp_action::Client<control_msgs::action::ParallelGripperCommand>::GoalHandle::SharedPtr&
+               /* unused-arg */) { RCLCPP_DEBUG_STREAM(logger_, name_ << " started execution."); };
+    // Send goal
+    auto current_goal_future = controller_action_client_->async_send_goal(goal, send_goal_options);
+    current_goal_ = current_goal_future.get();
+    if (!current_goal_)
+    {
+      RCLCPP_ERROR(logger_, "%s goal was rejected by server.", name_.c_str());
+      return false;
+    }
+
+    done_ = false;
+    last_exec_ = moveit_controller_manager::ExecutionStatus::RUNNING;
+    return true;
+  }
+
+  void setCommandJoint(const std::string& name)
+  {
+    command_joints_.clear();
+    addCommandJoint(name);
+  }
+
+  void addCommandJoint(const std::string& name)
+  {
+    command_joints_.insert(name);
+  }
+
+  void allowFailure(bool allow)
+  {
+    allow_stalling_ = allow;
+  }
+
+private:
+  void controllerDoneCallback(
+      const rclcpp_action::ClientGoalHandle<control_msgs::action::ParallelGripperCommand>::WrappedResult& wrapped_result)
+      override
+  {
+    if (wrapped_result.code == rclcpp_action::ResultCode::ABORTED && wrapped_result.result->stalled && allow_stalling_)
+    {
+      finishControllerExecution(rclcpp_action::ResultCode::SUCCEEDED);
+    }
+    else
+    {
+      finishControllerExecution(wrapped_result.code);
+    }
+  }
+
+  /*
+   * Some gripper drivers may indicate a failure if they do not close all the way when
+   * an object is in the gripper.
+   * If true, this assumes success if the gripper stalls and does not reach the target position goal.
+   */
+  bool allow_stalling_;
+
+  /*
+   * The ``max_effort`` used in the ParallelGripperCommand message.
+   */
+  double max_effort_ = 0.0;
+
+  /*
+   * The ``max_velocity_`` used in the ParallelGripperCommand message.
+   */
+  double max_velocity_ = 0.0;
+
+  /*
+   * The joints to command in the ParallelGripperCommand message
+   */
+  std::set<std::string> command_joints_;
```

Comment:
Do we actually need to have it as a set? Looking at https://github.com/ros-controls/ros2_controllers/blob/master/parallel_gripper_controller/src/gripper_action_controller_parameters.yaml user can only specify one joint. It will simplify the rest of the code as well. Happy to help with refactoring